src/chain.cpp

```diff
@@ -58,6 +58,13 @@ const CBlockIndex *CChain::FindFork(const CBlockIndex *pindex) const {
     return pindex;
 }
 
+CBlockIndex *CChain::FindLatestBefore(int64_t nTime) const {
+    CBlockIndex *pindex = Tip();
+    while (pindex && pindex->pprev && pindex->GetBlockTime() > nTime)
```

Comment:
As already commented, i think we should go down serval blocks (~144) because of possible inaccurate timestamps.